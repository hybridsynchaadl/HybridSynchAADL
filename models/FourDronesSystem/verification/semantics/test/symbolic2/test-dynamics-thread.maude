
set show timing off .

load ../../interpreter-symbolic2.maude
load module-component-dynamics.maude

select TEST-THREAD-DYNAMICS-MOD .

red finalConst(obj1) .

red finalConst(obj2) .

search executeStep(bool(true), transIn(r(4) # bool(true), c1 .. inp) obj1) =>* C:ConstConfig .

search executeStep(bool(true), transIn(r(5) # bool(true), c2 .. inp) obj2) =>* C:ConstConfig .

search executeStep(bool(true), transIn(r(5) # bool(true), c3 .. inp) obj3) =>* C:ConstConfig .

search executeStep(bool(true), transIn(r(5) # bool(true), c4 .. inp) obj4) =>* C:ConstConfig .
