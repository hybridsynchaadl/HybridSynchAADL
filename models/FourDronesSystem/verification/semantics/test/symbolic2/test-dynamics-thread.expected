==========================================
reduce in TEST-THREAD-DYNAMICS-MOD : finalConst(obj1) .
rewrites: 34
result SMTBoolValue: [(true).Boolean]
==========================================
reduce in TEST-THREAD-DYNAMICS-MOD : finalConst(obj2) .
rewrites: 34
result SMTBoolValue: [(true).Boolean]
==========================================
search in TEST-THREAD-DYNAMICS-MOD : executeStep(bool(true), obj1 transIn(r(4)
    # bool(true), c1 .. inp)) =>* C:ConstConfig .

Solution 1 (state 2)
states: 3  rewrites: 256
C:ConstConfig --> [not r("#$#1") <= (-1/1).Real + r("#$#4")] || < c1 : Thread |
    properties : none,features : (< inp : DataInPort | properties : none,
    content : [(0/1).Real] # [(false).Boolean],cache : [r("#$#4")] # [(
    true).Boolean] > < outp : DataOutPort | properties : none,content : [(
    1/1).Real + ((1/1).Real + r("#$#4"))] # [(true).Boolean] >),subcomponents :
    (< x : Data | properties : none,features : none,subcomponents : none,
    connections : empty,value : [(1/1).Real + r("#$#1")] # [(true).Boolean] > <
    y : Data | properties : none,features : none,subcomponents : none,
    connections : empty,value : [r("#$#4")] # [(true).Boolean] >),connections :
    empty,varGen : < "c1",1,2 >,currState : loc([(0/1).Real]),completeStates :
    loc([(0/1).Real]),variables : ((a : Real) ; b : Real),transitions : ((loc([
    (0/1).Real]) -[on dispatch]-> loc([(1/1).Real]) {(c{x} := (c[x] + [[(
    1).NzNat]])) ; (v{a} := f[inp]) ; c{y} := v[a]}) ; loc([(1/1).Real]) -[c[x]
    > c[y]]-> loc([(0/1).Real]) {(v{b} := (c[y] + [[(1).NzNat]])) ; f{outp} :=
    (v[b] + [[(1).NzNat]])}),loopBound : 10,transBound : 10 >

No more solutions.
states: 3  rewrites: 256
==========================================
search in TEST-THREAD-DYNAMICS-MOD : executeStep(bool(true), obj2 transIn(r(5)
    # bool(true), c2 .. inp)) =>* C:ConstConfig .

Solution 1 (state 4)
states: 5  rewrites: 289
C:ConstConfig --> [not r("#$#5") <= (0/1).Real and r("#$#5") <= (4/1).Real] ||
    < c2 : Thread | properties : none,features : (< inp : DataInPort |
    properties : none,content : [(0/1).Real] # [(false).Boolean],cache : [r(
    "#$#5")] # [(true).Boolean] > < outp : DataOutPort | properties : none,
    content : [(1/2).Real * r("#$#5") - (2/1).Real] # [(true).Boolean] >),
    subcomponents : < z : Data | properties : none,features : none,
    subcomponents : none,connections : empty,value : [(1/2).Real * r("#$#5")] #
    [(true).Boolean] >,connections : empty,varGen : < "c2",10,20 >,currState :
    loc([(2/1).Real]),completeStates : (loc([(0/1).Real]) loc([(2/1).Real])),
    variables : a : Real,transitions : ((loc([(0/1).Real]) -[on dispatch]->
    loc([(1/1).Real]) {(v{a} := f[inp]) ; if(v[a] > [[(0).Zero]])c{z} := (v[a]
    * [[5.0e-1]]) end if}) ; (loc([(1/1).Real]) -[otherwise]-> loc([(
    2/1).Real]) {f{outp} := (c[z] - [[(2).NzNat]])}) ; (loc([(1/1).Real]) -[c[
    z] > [[(2).NzNat]]]-> loc([(2/1).Real]) {f{outp} := (c[z] + [[(
    1).NzNat]])}) ; loc([(2/1).Real]) -[on dispatch]-> loc([(0/1).Real]) {
    skip}),loopBound : 10,transBound : 10 >

Solution 2 (state 5)
states: 6  rewrites: 363
C:ConstConfig --> [not r("#$#5") <= (0/1).Real and not r("#$#5") <= (4/1).Real]
    || < c2 : Thread | properties : none,features : (< inp : DataInPort |
    properties : none,content : [(0/1).Real] # [(false).Boolean],cache : [r(
    "#$#5")] # [(true).Boolean] > < outp : DataOutPort | properties : none,
    content : [(1/1).Real + (1/2).Real * r("#$#5")] # [(true).Boolean] >),
    subcomponents : < z : Data | properties : none,features : none,
    subcomponents : none,connections : empty,value : [(1/2).Real * r("#$#5")] #
    [(true).Boolean] >,connections : empty,varGen : < "c2",10,20 >,currState :
    loc([(2/1).Real]),completeStates : (loc([(0/1).Real]) loc([(2/1).Real])),
    variables : a : Real,transitions : ((loc([(0/1).Real]) -[on dispatch]->
    loc([(1/1).Real]) {(v{a} := f[inp]) ; if(v[a] > [[(0).Zero]])c{z} := (v[a]
    * [[5.0e-1]]) end if}) ; (loc([(1/1).Real]) -[otherwise]-> loc([(
    2/1).Real]) {f{outp} := (c[z] - [[(2).NzNat]])}) ; (loc([(1/1).Real]) -[c[
    z] > [[(2).NzNat]]]-> loc([(2/1).Real]) {f{outp} := (c[z] + [[(
    1).NzNat]])}) ; loc([(2/1).Real]) -[on dispatch]-> loc([(0/1).Real]) {
    skip}),loopBound : 10,transBound : 10 >

Solution 3 (state 6)
states: 7  rewrites: 452
C:ConstConfig --> [r("#$#3") <= (2/1).Real and r("#$#5") <= (0/1).Real] || < c2
    : Thread | properties : none,features : (< inp : DataInPort | properties :
    none,content : [(0/1).Real] # [(false).Boolean],cache : [r("#$#5")] # [(
    true).Boolean] > < outp : DataOutPort | properties : none,content : [r(
    "#$#3") - (2/1).Real] # [(true).Boolean] >),subcomponents : < z : Data |
    properties : none,features : none,subcomponents : none,connections : empty,
    value : [r("#$#3")] # [(true).Boolean] >,connections : empty,varGen : <
    "c2",10,20 >,currState : loc([(2/1).Real]),completeStates : (loc([(
    0/1).Real]) loc([(2/1).Real])),variables : a : Real,transitions : ((loc([(
    0/1).Real]) -[on dispatch]-> loc([(1/1).Real]) {(v{a} := f[inp]) ; if(v[a]
    > [[(0).Zero]])c{z} := (v[a] * [[5.0e-1]]) end if}) ; (loc([(1/1).Real]) -[
    otherwise]-> loc([(2/1).Real]) {f{outp} := (c[z] - [[(2).NzNat]])}) ; (loc(
    [(1/1).Real]) -[c[z] > [[(2).NzNat]]]-> loc([(2/1).Real]) {f{outp} := (c[z]
    + [[(1).NzNat]])}) ; loc([(2/1).Real]) -[on dispatch]-> loc([(0/1).Real]) {
    skip}),loopBound : 10,transBound : 10 >

Solution 4 (state 7)
states: 8  rewrites: 526
C:ConstConfig --> [r("#$#5") <= (0/1).Real and not r("#$#3") <= (2/1).Real] ||
    < c2 : Thread | properties : none,features : (< inp : DataInPort |
    properties : none,content : [(0/1).Real] # [(false).Boolean],cache : [r(
    "#$#5")] # [(true).Boolean] > < outp : DataOutPort | properties : none,
    content : [(1/1).Real + r("#$#3")] # [(true).Boolean] >),subcomponents : <
    z : Data | properties : none,features : none,subcomponents : none,
    connections : empty,value : [r("#$#3")] # [(true).Boolean] >,connections :
    empty,varGen : < "c2",10,20 >,currState : loc([(2/1).Real]),completeStates
    : (loc([(0/1).Real]) loc([(2/1).Real])),variables : a : Real,transitions :
    ((loc([(0/1).Real]) -[on dispatch]-> loc([(1/1).Real]) {(v{a} := f[inp]) ;
    if(v[a] > [[(0).Zero]])c{z} := (v[a] * [[5.0e-1]]) end if}) ; (loc([(
    1/1).Real]) -[otherwise]-> loc([(2/1).Real]) {f{outp} := (c[z] - [[(
    2).NzNat]])}) ; (loc([(1/1).Real]) -[c[z] > [[(2).NzNat]]]-> loc([(
    2/1).Real]) {f{outp} := (c[z] + [[(1).NzNat]])}) ; loc([(2/1).Real]) -[on
    dispatch]-> loc([(0/1).Real]) {skip}),loopBound : 10,transBound : 10 >

No more solutions.
states: 8  rewrites: 530
Bye.
